src/Symfony/Component/Notifier/Bridge/Expo/ExpoTransport.php

```
@@ -0,0 +1,104 @@
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Imad ZAIRIG <https://github.com/zairigimad>
+ *
+ * @experimental in 5.4
+ */
+final class ExpoTransport extends AbstractTransport
+{
+    protected const HOST = 'exp.host/--/api/v2/push/send';
+
+    /** @var string */
+    private $token;
+
+    public function __construct(string $token = null, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->token = $token;
+        $this->client = $client;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('expo://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        $endpoint = sprintf('https://%s', $this->getEndpoint());
+        $options = ($opts = $message->getOptions()) ? $opts->toArray() : [];
+        if (!isset($options['to'])) {
+            $options['to'] = $message->getRecipientId();
+        }
+        if (null === $options['to']) {
+            throw new InvalidArgumentException(sprintf('The "%s" transport required the "to" option to be set.', __CLASS__));
+        }
+        $options['body'] = $message->getSubject();
+        $options['data'] = $options['data'] ?? [];
+
+        $response = $this->client->request('POST', $endpoint, [
+            'headers' => [
+                'Authorization' => $this->token ? sprintf('Bearer %s', $this->token) : null,
```

I think the API will return an error if there is no token, maybe you can throw an error and avoid the request.

---

I already tested it, and it works, if there is no token normally returns nothing in the header. it will not be taken into consideration.